### PR TITLE
ci: Adapted Makefile to argo

### PR DESCRIPTION
**Pull request by dgiebert**
Reworked the Makefile to be in similar to: https://github.com/argoproj/argo/blob/master/Makefile
Reworked the Dockerfile to be in sync with: https://github.com/argoproj/argo/blob/master/Dockerfile

Added a CI to push images: https://github.com/dgiebert/argo-events/runs/1011429452
Needed Secrets:
- DOCKERIO_USERNAME
- DOCKERIO_PASSWORD
- DOCKERIO_ORG

Automatically releases master and tags, see e.g. here: https://hub.docker.com/repository/docker/dgiebert/eventbus-controller

Closes: https://github.com/argoproj/argo-events/issues/836

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,44 +1,95 @@
 ####################################################################################################
-# base
+# Builder image
```
[the remark this answers is not in the document]
Comment (dgiebert) [reply Dockerfile:2]: I can add that back in

File: .github/workflows/docker-image.yml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+          if [ $tag = "master" ]; then
+            tag="latest"
+          fi
+          targets="eventbus-controller eventsource-controller sensor-controller eventsource sensor"
+          for target in $targets; do
+            docker buildx build \
+              --output "type=image,push=true" \
+              --platform="linux/amd64,linux/arm64" \
+              --target $target \
+              --tag "${DOCKERIO_ORG}/${target}:${tag}-linux" .
```
[the remark this answers is not in the document]
Comment (dgiebert) [reply .github/workflows/docker-image.yml]: I removed the -linux from the tag now with the latest commit.

And you are correct, master would be tagged latest currently

File: .github/workflows/docker-image.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
+
+name: Docker Image CI
+
+on:
+  push:
+    tags:
+      - v*
+    branches:
```
Comment (whynowy) [inline .github/workflows/docker-image.yml:8]: I think disabling `master` branch build might be better. It doesn't make sense to put `master` branch build as `latest`.

File: Makefile [hunk 4]
```diff
@@ -2,7 +2,7 @@ PACKAGE=github.com/argoproj/argo-events
 CURRENT_DIR=$(shell pwd)
 DIST_DIR=${CURRENT_DIR}/dist
 
-DOCKERFILE:=Dockerfile
+DOCKERFILE:=Dockerfile.dev
```
[the remark this answers is not in the document]
Comment (dgiebert) [reply Makefile:5]: I have added the amd64 target to the image creation

**Pull request merged**